documentation/library_guide/parallel_api/iterators_advanced_use_cases.rst

```diff
@@ -21,6 +21,42 @@ header.  All iterators are implemented in the ``oneapi::dpl`` namespace.
     auto sum = std::reduce(dpl::execution::dpcpp_default,
                            count_a, count_b, init); // sum is (0 + 0 + 1 + ... + 9) = 45
 
+* ``zip_iterator``
+
+  The ``zip_iterator`` is an iterator constructed with one or more iterators as input. The result of
+  ``zip_iterator`` dereferencing is a tuple-like object of unspecified type holding the values returned by
+  dereferencing the member iterators, which ``zip_iterator`` wraps. Arithmetic operations performed on a
+  ``zip_iterator`` instance are also applied to each of the member iterators.
+
+  The ``make_zip_iterator`` function is provided to simplify the construction of ``zip_iterator`` instances.
+  The function returns ``zip_iterator`` instance with all the agruments held as member iterators.
+
+  The ``zip_iterator`` is useful in defining by key algorithms where input iterators
+  representing keys and values are processed as key-value pairs. The example below demonstrates a stable sort
+  by key where only the keys are compared but both keys and values are swapped::
+
+    using namespace oneapi;
```

Comment:
Adding `.. code:: cpp` will highlight the syntax. However, I see that it's not used in other places of the file.